internal/command/stacks.go

```diff
@@ -0,0 +1,363 @@
+		Cmd:              exec.Command(c.pluginBinary),
+		Logger:           logging.NewStacksLogger(),
+		VersionedPlugins: map[int]plugin.PluginSet{
+			1: {
+				"stacks": &stacksplugin1.GRPCStacksPlugin{
+					Metadata: c.pluginConfig.ToMetadata(),
+					Services: c.Meta.Services,
+				},
+			},
+		},
+	})
+	defer client.Kill()
+
+	// Connect via RPC
+	rpcClient, err := client.Client()
+	if err != nil {
+		fmt.Fprintf(stderr, "Failed to create stacks plugin client: %s", err)
+		return ExitRPCError
+	}
+
+	// Request the plugin
+	raw, err := rpcClient.Dispense("stacks")
+	if err != nil {
+		fmt.Fprintf(stderr, "Failed to request stacks plugin interface: %s", err)
+		return ExitRPCError
+	}
+
+	// Proxy the request
+	// Note: future changes will need to determine the type of raw when
+	// multiple versions are possible.
+	stacks1, ok := raw.(pluginshared.CustomPluginClient)
+	if !ok {
+		c.Ui.Error("If more than one stacksplugin versions are available, they need to be added to the stacks command. This is a bug in Terraform.")
+		return ExitRPCError
+	}
+
+	return stacks1.Execute(args, stdout, stderr)
+}
+
+// discoverAndConfigure is an implementation detail of initPlugin. It fills in the
+// pluginService and pluginConfig fields on a StacksCommand struct.
+func (c *StacksCommand) discoverAndConfigure() tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+
+	// using the current terraform path for the plugin binary path
+	tfBinaryPath, err := os.Executable()
+	if err != nil {
+		return diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Terraform binary path not found",
+			"Terraform binary path not found: "+err.Error(),
+		))
+	}
+
+	// stacks plugin requires a cloud backend in order to work,
+	// however `cloud` block in not yet allowed in the stacks working directory
+	// initialize an empty cloud backend
+	bf := backendInit.Backend("cloud")
+	if bf == nil {
+		return diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"`cloud` backend not found, this should not happen",
+			"`cloud` backend is a valid backend type, yet it was not found, this is could be a bug, report it.",
+		))
+	}
+	b := bf()
+	cb, ok := b.(*cloud.Cloud)
+	if !ok {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"`cloud` backend could not be initialized",
+			"Could not initialize a `cloud` backend, this is could be a bug, report it.",
+		))
+		return diags
+	}
+
+	displayHostname := os.Getenv("TF_STACKS_HOSTNAME")
+	if strings.TrimSpace(displayHostname) == "" {
+		return diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"TF_STACKS_HOSTNAME is not set",
+			"TF_STACKS_HOSTNAME must be set to the hostname of the HCP Terraform instance",
+		))
+	}
+
+	token := os.Getenv("TF_STACKS_TOKEN")
+	if strings.TrimSpace(token) == "" {
+		return diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"TF_STACKS_TOKEN is not set",
+			"TF_STACKS_TOKEN must be set to the token of the HCP Terraform instance",
+		))
+	}
```

Comment:
Could we use the configured services to try to find the token here? Then have the env var as a backup if there isn't an entry for the given hostname. I think the credentials file is read in part of the creation of the services 🤔, correct me if I'm wrong or if it's difficult to ascertain in this circumstance.

Reply:
Good point, token are now read from cred files if not provided